homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -251,6 +262,13 @@ def device_state_attributes(self):
         """Return the state attributes of the sensor."""
         return {
             ATTR_OBSERVATIONS: list(self.current_obs.values()),
+            ATTR_ENTITY_ID: list(
```

Comment:
Why are we calculating this on demand from self.current_obs ? I would expect this list to contain all entities that we are observing, not be the ones that we're currently observing. So I would expect us to calculate this once inside `__init__` based on `self.entity_obs_dict`

Reply:
ATTR_OBSERVATIONS contains values of entities that are observed. 
ATTR_ENTITY_ID should contain the entities that are observed. This changes with time and allows to have an overview of entities that are contributing to the status of the sensor.

Reply:
`ATTR_ENTITY_ID` is reserved to contain all entities that are part of something. Groups etc. So it should be ALL entities that are observed. If you want to have an attribute for ones that are contributing values currently, you should pick a different attribute.